### PR TITLE
Publish the changelog from default branch

### LINKED ISSUE #523: Changelogs for bugfix releases are not immediately published
This is a transformation of https://pulp.plan.io/issues/9361

### PR DESCRIPTION
**Pull request by mdellweg**
This will automatically publish the changelog to the docs root directory
whenever a changelog cherry pick is merged.

fixes #523

### CODE REVIEW

**Comment by mdellweg** [comment]
I don't yet know, how to test this.

**Comment by mdellweg** [comment]
The building part seems to work now.

**Comment by bmbouter** [comment]
@dkliban can you remind me of the layout of the docs on pulpproject.org ? I'm wondering specifically if we have "nightly" for each version of pulpcore for example.

**Comment by bmbouter** [comment]
@mdellweg is the goal of this PR to push docs with every merge (pre-release right?) I was thinking yes because the release automation handles the pushing in other cases already (is that right?)

**Comment by mdellweg** [comment]
This PR intends to target the auto-generated "cherry-pick changelog" PR's specifically. On merge, they change the `CHANGES.rst` on the main branch, and that is the trigger for the new workflow. It does not touch any of the docs in any sub-directory, but only updates the most prominent place of the changelog.

**Comment by bmbouter** [comment]
I see, so the idea is when a say 3.15.z changelog is merged to master, also push it to docs.pulpproject.org/latest/ is that right?

If my understanding is right, personally I don't think the solution and maintaining it is worth the benefit. The changelog will be right the next time a release happens, and this is getting complex. That's my take at least.

**Comment by mdellweg** [comment]
> The changelog will be right the next time a release happens

That is exactly the problem that was reported multiple times.

**Comment by bmbouter** [comment]
> > The changelog will be right the next time a release happens
> 
> That is exactly the problem that was reported multiple times.

Yup and they're right, but to me it's not worth the complexity. If others want to review it and go forward with this, I'm ok with that.

File: templates/github/.github/workflows/changelog.yml.j2 [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+with context %}
+---
+name: {{ plugin_camel_short | default("Pulp") }} changelog update
+on:
+  push:
+    branches:
+      - {{ plugin_default_branch }}
+    paths:
+      - CHANGES.rst
+      - CHANGES.md
```
Comment (fao89) [inline templates/github/.github/workflows/changelog.yml.j2:16]: I think it should be:
```suggestion
      - CHANGES/**
```

Comment (mdellweg) [reply templates/github/.github/workflows/changelog.yml.j2:16]: I don't think so. We want to republish the generated changelog (either rst or md) and not every time something happened with the "unreleased" snippet files.

Comment (fao89) [reply templates/github/.github/workflows/changelog.yml.j2:16]: Oh! I misunderstood the PR, I thought it was for building changelogs whenever a new entry was added.

Comment (mdellweg) [reply templates/github/.github/workflows/changelog.yml.j2:16]: No, it's publishing the complete changelog whenever it was changed.

**Review by fao89 — APPROVED** [review]
approved but it needs a rebase

**Review by fao89 — APPROVED** [review]
LGTM

**Review by dkliban — APPROVED** [review]

**Pull request merged**